### ISSUE TITLE
Panic when import cluster that has nodegroups using a launch template

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#28801]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. create cluster with eksctl that has one nodegroup

**Result:**
panic observed in eks-operator logs

**Other details that may be helpful:**
There will be another issue for further support of launch templates.
EDIT: https://github.com/rancher/rancher/issues/28803

**Comment by aaronRancher** [comment]
The bug is reproduced with Rancher server `rancher\rancher:v2.5.0-alpha1`

Steps:
Create an eks cluster using eksctl: `eksctl create cluster -f managed-cluster.yaml`
This cluster will have a node group with a default launch template.
![Screen Shot 2020-09-14 at 6 00 48 PM](https://user-images.githubusercontent.com/57409052/93152733-504d2280-f6b4-11ea-9b3f-44efd28756e0.png)

YAML:
```
# managed-cluster.yaml
# A cluster with two managed nodegroups
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: aaron-managed-cluster4
  region: us-west-2

managedNodeGroups:
  - name: managed-ng-1
    minSize: 2
    desiredCapacity: 2
    maxSize: 4
    labels:
      role: worker

```
Import cluster into Rancher.

Result:
Cluster is stuck provisioning
Observed panic in eks-operator logs
![Screen Shot 2020-09-14 at 5 46 40 PM](https://user-images.githubusercontent.com/57409052/93152744-5cd17b00-f6b4-11ea-9c23-47c96a430f8e.png)

**Comment by sangeethah** [comment]
Reopening for the attached PR to get merged

**Comment by cbron** [comment]
Wrong PR was linked, this is actually ready for test.

**Comment by sangeethah** [comment]
@cbron We are blocked on import testing - https://github.com/rancher/rancher/issues/28900

**Comment by cbron** [comment]
@sangeethah should be unblocked now

**Comment by aaronRancher** [comment]
The bug fix is verified with Rancher server `rancher\rancher:master-head` `version 53cf961`

Steps:
Create an eks cluster using eksct.
This cluster will have a node group with a default launch template.
Import cluster into Rancher.

Result:
Cluster comes into active state successfully.
No panic seen in eks-operator:
```
W0916 20:54:21.888282       1 loader.go:223] Config not found: /root/.kube/config
time="2020-09-16T20:54:22Z" level=info msg="Starting eks.cattle.io/v1, Kind=EKSClusterConfig controller"
time="2020-09-16T20:54:22Z" level=info msg="Starting /v1, Kind=Secret controller"
time="2020-09-16T20:54:22Z" level=error msg="Error recording ekscc [] failure message: resource name may not be empty"
time="2020-09-16T20:54:22Z" level=info msg="cluster [c-59zrs] finished updating"
time="2020-09-16T20:54:22Z" level=error msg="Error recording ekscc [] failure message: resource name may not be empty"
time="2020-09-16T20:54:23Z" level=info msg="cluster [c-59zrs] finished updating"
time="2020-09-16T20:54:23Z" level=info msg="cluster [c-59zrs] finished updating"
```
